feat: a simproc version of `compute_degree`

5hv5hvnk opened the pull request:
Wrap `compute_degree` in a simproc for use by simp.

Closes #22219.

---

t-5hv5hvnk committed:
added test and PolynomialDegree

YaelDillies reviewed (commented):
Thank you for taking this up 🎉 

I don't understand: Are you reimplementing `compute_degree`. My idea when writing #22219 was to reuse the `compute_degree` internals

5hv5hvnk committed:
Merge branch 'leanprover-community:master' into computer_degree2simproc

5hv5hvnk committed:
Update Mathlib/Tactic/Simproc/PolynomialDegree.lean

Co-authored-by: Yaël Dillies <yael.dillies@gmail.com>

t-5hv5hvnk committed:
used ComputeDegree from mathlib

t-5hv5hvnk committed:
t

Merge branch 'computer_degree2simproc' of https://github.com/5hv5hvnk/mathlib4 into computer_degree2simproc

YaelDillies reviewed (commented):
Ahah, I was in fact writing a review already.

@adomani, are you interested in reviewing this?

adomani reviewed (commented):
I simply left some really minor comments.

CI is currently complaining about the fact that you should commit the files that `lake exe mk_all --git` produces.  This is simply to make sure that `Mathlib.lean` and `Mathlib/Tactic.lean` stay up to date.

However, when I open the test file locally, it just consists of error messages.  Even adding a blanket `open Polynomial` does not fix the issues.
